### PR TITLE
chore(rome_*):reduce parking_lot usage

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
1. Since we have upgraded `rust-toolchain` to **1.62.0 stable** which has been stabilized the new `Mutex` api.
Also, it is faster than `parking_lot` in Linux platform https://blog.rust-lang.org/2022/06/30/Rust-1.62.0.html#thinner-faster-mutexes-on-linux.
So, we could reduce the `parking-lot` crate usage to speed up our compile-time a little.

## Test Plan
1. cargo test

### CODE REVIEW

**Comment by xunilrj** [comment]
Will be interesting to see the differences.

**Comment by IWANABETHATGUY** [comment]
> 

I don't benchmark this pull request, You could see the detailed benchmark made by rust core team https://github.com/rust-lang/rust/pull/95035#issuecomment-1073966631

**Comment by IWANABETHATGUY** [comment]
This pull request more likely to be a compile-time improvement

File: crates/rome_js_formatter/Cargo.toml [hunk 1]
```diff
@@ -23,13 +23,13 @@ tests_macros = { path = "../tests_macros" }
...
 rome_diagnostics = { path = "../rome_diagnostics" }
 countme = { version = "3.0.0", features = ["enable"] }
-parking_lot = "0.12.0"
 similar = "2.1.0"
 similar-asserts = "1.2.0"
 ctor = "0.1.21"
 quickcheck = "1.0.3"
 quickcheck_macros = "1.0.0"
 iai = "0.1.1"
+parking_lot = "0.12.0"
```
Comment (ematipico) [inline crates/rome_js_formatter/Cargo.toml:32]: I am not sure what's this change for.

**Review by ematipico — CHANGES_REQUESTED** [review]
I know it's annoying, but each `unwrap` should be documented with a `// SAFETY` comment

**Comment by IWANABETHATGUY** [comment]
The const *constructor* of `{Mutex, Condvar, RwLock}`  would be stable until 1.63, https://github.com/rust-lang/rust/pull/97791/files#diff-3ec0c54e5f945521d425d023283b935edaa5c16fe7e2c0f675b07f35e65f639bR125, So i am going to close this pr for now

**Pull request closed without merging**